File: 209. Minimum Size Subarray Sum/TwoPtr.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        
        int MinSize = nums.size() + 1;
        int SumBuff = 0;
        for(int Left = 0,  Right = 0; Right < nums.size(); ++Right){

            SumBuff += nums[Right];

            for(; SumBuff >= target; ++Left){

                MinSize = min(MinSize, Right- Left + 1);
                SumBuff -= nums[Left];
            }
        }
        
        return MinSize == nums.size() + 1 ? 0 : MinSize;

    }
};

File: 167. Two Sum II - Input Array Is Sorted/two_ptr.go
func twoSum(nums []int, target int) []int {
	L, R := 0, len(nums)-1

	for target != nums[L]+nums[R] {
		if target < nums[L]+nums[R] {
			R--
		} else {
			L++
		}
	}

	return []int{L + 1, R + 1}
}
